### DISCUSSION TITLE
Request for a developer feature to add new fields in layout settings model for specific field type

### DISCUSSION THREAD

**Discussion by amici-infotech** [craftcms/cms#13491] [Ideas]
Hey. I have a custom field type, I wanted to add a custom settings fields here in the layout settings modal. is that possible? if not, can you please consider it as a feature request?

![image](https://github.com/craftcms/cms/assets/20450552/c4dfec62-dc65-4df0-b38a-5bdd818ca362)

**Comment by thupsi** [comment]
Can you share an example of what you are trying to achieve?

**Comment by amici-infotech** [comment]
Basically, we recieved a FR (https://github.com/amici-infotech/craft-super-dynamic-fields/issues/7) where users wants to select different JSON templates for options on basis of entry types.

![image](https://github.com/craftcms/cms/assets/20450552/8a74e966-8a01-42b6-9c81-39a832fc03e9)

**Reply by thupsi** [reply]
Now I see what you mean, and yes it would be usefull for that specific case. 

It seems to me though that bending the rules of what parts of a field are configurable on an entry type level would potentialy introduce complications, but I'd better let someone from the team comment on that! 🙂

Edit: I like the workaround that you proposed for the issue! Seems pretty clean to me!
